### ISSUE TITLE
Kontakt 6 preset loading issue

### ISSUE THREAD

**Issue by Nintorac** [falkTX/Carla#1053]
Loading a Kontakt patch from a `carxp` causes the plugin to freeze. However if a plugin is first loaded and then the state is loaded using a `carxs` and the `Load State` button the patch loads successfully.

During a failed load of Kontakt I can see callbacks come in for some parameters, however when loading a patch successfully the same callbacks come in, followed by a pause and then an extra set of parameter change callbacks are made. This makes me believe that Kontakt has a two stage loading process that is somehow blocked when the plugin is loaded directly from a `carxp`

I have tried to do a little digging and the difference between loading state and loading project seems to be the enabled status of the plugin. Unfortunately going deeper is a bit beyond my abilities at this point

https://github.com/falkTX/Carla/blob/develop/source/backend/engine/CarlaEngine.cpp#L2723

https://github.com/falkTX/Carla/blob/6b2f607169d0c58c2a5e7fcda9f993a71b9a81f6/source/backend/plugin/CarlaPlugin.cpp#L996

**Comment by falkTX** [comment]
Is this under linux or windows?
Does the same happen with the Kontakt player plugin? Just so I can try to reproduce the issue here

**Comment by Nintorac** [comment]
This is under linux

**Comment by Laguna-x** [comment]
may be same issue with Kontakt 5 here:
```
 carla -v
Using Carla version 2.0.97 (2.1-RC2)
  Python version: 3.7.5
  Qt version:     5.12.4
  PyQt version:   5.12.3
  Binary dir:     /usr/lib/carla
  Resources dir:  /usr/share/carla/resources

uname -a
Linux Media 5.3.0-40-lowlatency #32-Ubuntu SMP PREEMPT Fri Jan 31 21:48:22 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

apt search carla
carla/bionic 5:2.0.0+git20190808.3 amd64
carla-git/bionic,now 5:2.1~rc1+git20200331 amd64 [установлен, автоматически]
carla-data/bionic,bionic 5:2.0.0+git20190808.3 all
carla-git-data/bionic,bionic,now 5:2.1~rc1+git20200331 all [установлен, автоматически]
carla-vst/eoan-updates 2.0.0-0ubuntu4 i386
carla-vst-wine/bionic,now 5:2.1~rc1+git20200331 amd64 [установлен]

```
trying to load saved config with Kontakt library and loading library stuck.
1) create new preset
2) add Kontakt 5
4) <saving project> <closing carla> <starting carla> <loading project> - no any errors or anything else bad
5)  load library in Kontakt
6) playing all well
7) <saving project> <closing carla> <starting carla> <loading project> - loading Kontakt with stuck and no error-messages.
8) <delete Kontakt><add Kontakt><load any library> - all works well. but repeat step 7 makes same stuck as in end of step 7

**Comment by moonshaadow** [comment]
There's still some bugs using Kontakt under Wine. Kontakt 6 works better using last versions of wineHQ. There some steps to make, following instructions [here](https://github.com/osxmidi/LinVst/tree/master/Tested-VST-Plugins) is a good start. But some details won't works, often gui-related. Don't try to drag/drop, to use internal Kontakt way to load patches is preferred.

**Comment by lagunax** [comment]
found workaround:
befor saving settings, if kontakt will be switched off (top left button of plugin line in carla rack), then it loads fast without any problems next time.
if save with switched on - then carla stucks on loading kontakt data with any settings of carla.

**Comment by moonshaadow** [comment]
Hello there.
@Nintorac : 
I can reproduce your bug , but only in some case. Typically  those when the gui is waiting an user's action. (replace multi, missing samples...). Disable the option "show 'replace multi' dialog" in K6 could help if you use .nkm in your project.
There is a wine bug affecting Kontakt and some other apps : pop-up windows are ok when resulting from an action in kontakt, but are unavailable or freeze when resulting from an action in linux environment. See [(https://bugs.winehq.org/show_bug.cgi?id=48995) ] for a patch.

**Comment by moonshaadow** [comment]
Hi.
(Sorry for my last comment, this is not related to the wine bug I have mentioned.)
I have experimented same bug , with K6 and K5, last wine-staging.
@falkTX , the bug is _not happening_ with carla 2.0.0 + your wine-rt. Not happening with every Kontakt banks, so this could not be easy to reproduce. 
Work-around by @lagunax is working.

**Comment by P-Mousavi** [comment]
I think this might help:
I get `Carla assertion failure: "childPID == 0" in file threads/ChildProcess.cpp, line 154, value 28384` when closing Ardour6 after working with Kontakt6 using Carla2.2.And then I have to load the instruments manually next time I open my session in Ardour.

**Comment by falkTX** [comment]
That makes me think the bridge-side is crashing, so that "pid == 0" (that is, process was no longer running when we attempted to stop it)

**Comment by moonshaadow** [comment]
@falkTX  Some news on this issue ?
Currently it's not possible to load a project with an active Kontakt plugin : the plugin freeze at starting the loading of samples.

I think about a bridge crash too since I've performed the following tests : 
- Kontakt in a wine host is ok
- Kontakt in carla through yabridge is ok
- Kontakt in carla for Windows is ok

**Comment by moonshaadow** [comment]
Here the logs :
```
Carla assertion failure: "Z:\usr\lib\carla\carla-bridge-native.exe" in file files/File.cpp, line 180

CarlaRingBuffer::tryWrite(0x7ffff8ceb9b4, 4): failed, not enough space
Carla Bridge Ready!
```

**Comment by falkTX** [comment]
This will be hard for me to do anything about since I do not have access to the plugin.
Is there a free kontakt "player" ? maybe it has the same issues...

**Comment by moonshaadow** [comment]
@falkTX Yes there is a free kontakt player, but you have to install it via Native Access, known to have issues with wine. 
[Native Access](https://www.native-instruments.com/en/specials/native-access/)
I could also try to help if you tell me how.

**Comment by falkTX** [comment]
I would try to debug where the stalling is happening exactly, meaning putting some prints around the code and attempting to load, repeating until it is clear where the issue is.

**Comment by moonshaadow** [comment]
@falkTX It seems a bit out of my abilities to use prints in your code for now.
But here is what I've found : 
- A bug in loading Kontakt projects appears starting commit 7b754358e6ee2a204c7ed24a56d465f3e4a1fdd2.
   At this point, the bug is : When loading .carxp, plugin starts but can't load project. It results in an empty Kontakt, still usable.
  We can suppose here that there is a problem with Juce stuff.
- The bug changes starting commit fea4a0b12da03287a13848722feb8a2698f5514c.
   Now the bug is : When loading .carxp, plugin fails to load, most of the time "freezing" on a modal window. The pointer disappear on the window so we cannot interact with it.
![kontaktt_freeze](https://user-images.githubusercontent.com/29754932/111088352-15bf2800-8527-11eb-974a-4c5dc8e605fb.png)
  At this point Carla seems to consider the plugin loaded, but Kontakt is unusable. Concretely, the modal window indicates that a sample library is loading.
